src/py-opentimelineio/opentimelineio/console/otiopluginfo.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python
+#
+# copyright 2019 pixar animation studios
+#
+# licensed under the apache license, version 2.0 (the "apache license")
+# with the following modification; you may not use this file except in
+# compliance with the apache license and the following modification to it:
+# section 6. trademarks. is deleted and replaced with:
+#
+# 6. trademarks. this license does not grant permission to use the trade
+#    names, trademarks, service marks, or product names of the licensor
+#    and its affiliates, except as required to comply with section 4(c) of
+#    the license and to reproduce the content of the notice file.
+#
+# you may obtain a copy of the apache license at
+#
+#     http://www.apache.org/licenses/license-2.0
+#
+# unless required by applicable law or agreed to in writing, software
+# distributed under the apache license with the above modification is
+# distributed on an "as is" basis, without warranties or conditions of any
+# kind, either express or implied. see the apache license for the specific
+# language governing permissions and limitations under the apache license.
+#
+
+"""Print information about the OTIO plugin ecosystem."""
+
+import argparse
+import fnmatch
+import textwrap
+import opentimelineio as otio
+
+
+OTIO_PLUGIN_TYPES = ['all'] + otio.plugins.manifest.OTIO_PLUGIN_TYPES
+
+
+def _parsed_args():
+    """ parse commandline arguments with argparse """
+
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument(
+        '-p',
+        '--plugin-types',
+        type=str,
+        default='all',
+        nargs='+',
+        choices=OTIO_PLUGIN_TYPES,
```

Comment:
This is cool, but the help output is a little strange here: 
```
optional arguments:
  -h, --help            show this help message and exit
  -p {all,adapters,media_linkers,schemadefs,hook_scripts,hooks} [{all,adapters,media_linkers,schemadefs,hook_scripts,hooks} ...], --plugin-types {all,adapters,media_linkers,schemadefs,hook_scripts,hooks} [{all,adapters,media_linkers,schemadefs,hook_scripts,hooks} ...]
```
I wonder if something could be done to clean that up a little via argparse.

Reply:
Hm... not sure I have much of a choice here unfortunately.  That seems to be how they format arguments with multiple options.  
https://docs.python.org/3/library/argparse.html#choices

If you can find an alternative way, I'm open to suggestions.